File: leetcode/MajorityElements.go
func majorityElement(nums []int) (candidate int) {
	cnt := 0
	for _, num := range nums {
		if cnt == 0 {
			candidate = num
			cnt++
		} else {
			if candidate == num {
				cnt++
			} else {
				cnt--
			}
		}
	}
	return
}

File: leetcode/MajorityElements.js
var majorityElement = function(nums) {
    
    let majorityElement = 0, counter = 0;
    
    for(let num of nums){
        if(counter == 0) majorityElement = num;
        majorityElement == num ? counter++: counter--;
    }
    
    return majorityElement;
};
